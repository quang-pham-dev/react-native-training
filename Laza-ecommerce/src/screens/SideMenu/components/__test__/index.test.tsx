import React from 'react';
import { Pressable } from 'react-native';

// LIBS
import { fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

// Screen
import DrawerItem from 'screens/SideMenu/components/index';

describe('Side menu', () => {
  let tree: any;
  const closeMenuID = 'Side-menu-close';

  const props = {
    title: 'title',
    onPress: jest.fn()
  };
  tree = renderer.create(<DrawerItem {...props} source={0} />);

  test('should render correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should call function handlePressMenuItem', async () => {
    const button = tree.root.findAllByType(Pressable)[0];

    await fireEvent.press(button);

    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
