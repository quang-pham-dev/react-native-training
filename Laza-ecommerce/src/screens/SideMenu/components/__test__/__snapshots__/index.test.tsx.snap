// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side menu should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 25,
      }
    }
  >
    <Image
      source={0}
      style={
        Object {
          "height": 19,
          "marginRight": 13,
          "resizeMode": "contain",
          "width": 19,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Inter_400Regular",
            "fontSize": 15,
            "lineHeight": 16,
          },
          undefined,
        ]
      }
    >
      title
    </Text>
  </View>
</View>
`;
